src/corecel/math/ConvexHullFinder.hh

```diff
@@ -0,0 +1,303 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/math/ConvexHullFinder.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <algorithm>
+
+#include "corecel/Assert.hh"
+#include "corecel/cont/Range.hh"
+#include "corecel/math/SoftEqual.hh"
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+/*!
+ * SFINAE struct for determing if a templated type is a valid Real2.
+ */
+template<typename T>
```

Comment:
@sethrj this works everywhere but Windows. There are at least 4 reasonable ways of representing a Real2, so I want to keep this templated, but since is a non-POD type it would be nice to have a check like this.

Reply:
#Since the only R2 we care about (I think??) is `celeritas::Array<T, 2>` why not template on just T?

Reply:
That's fine by me. I wanted to support std::vector as well but there isn't that good a reason for it.